diesel_derives/src/sql_function.rs

```diff
@@ -123,7 +125,39 @@ pub(crate) fn expand(input: SqlFunctionDecl) -> Result<TokenStream, Diagnostic>
         }
     };
 
-    if !is_aggregate {
+    if is_aggregate {
+        if cfg!(feature = "sqlite") {
+            let type_params = &generics
+                .type_params()
+                .filter(|type_param| !type_param.bounds.is_empty())
+                .cloned()
+                .collect::<Vec<_>>();
+
+            tokens = quote! {
+                #tokens
+
+                use diesel::sqlite::{Sqlite, SqliteConnection};
+                use diesel::serialize::ToSql;
+                use diesel::deserialize::Queryable;
+                use diesel::sqlite::Aggregator;
+
+                #[allow(dead_code)]
+                pub fn register_impl<A, #(#arg_name),*, #(#type_args),*>(
```

Comment:
Ideally, our `register_impl` would only require specifying the Aggregator trait implementation, but this requires a lot more type params. Is there any way around this that you can think of i.e. am I approaching this macro incorrectly?